### PR TITLE
[SDP basic tests] Add dma buf support

### PR DESCRIPTION
**Pull request by omasse-linaro**
Use DMABUF instead of Ion for Kernel version greater than 5.11

Signed-off-by: Olivier Masse <olivier.masse@nxp.com>

### CODE REVIEW

**Review by etienne-lms — COMMENTED** [review]
LGTM with few minor comments.

**Comment by omasse-linaro** [comment]
Tested on Hikey960

File: host/xtest/sdp_basic.h [hunk 1]
```diff
@@ -19,11 +29,19 @@ enum test_target_ta {
 	TEST_TA_TO_PTA,
 };
 
-int allocate_ion_buffer(size_t size, int heap_id, int verbosity);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
+int allocate_dma_buffer(size_t size, const char *heap_name, int heap_id, int verbosity);
+#define allocate_buffer(size, heap_name, heap_id, verbosity) \
```
[the remark this answers is not in the document]
Comment (omasse-linaro) [reply host/xtest/sdp_basic.h]: done here 86cd944

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by jforissier** [comment]
@omasse-linaro please add Etienne's A-b, replace "dma buf" with "dmabuf" in the subject and commit description (it looks like it is the preferred way to spell it). And also replace "[SDP basic tests]" by "sdp:". Thanks!

**Comment by jforissier** [comment]
Removed blank line between A-b and S-o-b and merged manually. Thanks @omasse-linaro.

**Pull request closed without merging**